### ISSUE TITLE
XM: Incorrect abort after write error on disk in receive mode

### ISSUE THREAD

**Issue by hubertushirsch** [wwarthen/RomWBW#452]
Hello,

when porting the XM utility (file xmdm125.asm in 3.5.0-dev.91) to my own operating system, I discovered the following error:

If a write error occurs when writing the buffer to disk, the program jumps to the WRERR label.
There a CAN is sent to the transmitter.
However, the CAN character must be loaded into the A register, not into register C.
```
WRERR:  CALL	RSDMA		; RESET DMA TO NORMAL
        MVI	C,CAN		; CANCEL
   	CALL	SEND		; SENDER
```
Should be
```
WRERR:	CALL	RSDMA		; RESET DMA TO NORMAL
 	MVI	A,CAN		; CANCEL
	CALL	SEND		; SENDER
```
I also found that the CAN must be sent at least three times for it to abort.
If only one or two CAN characters are sent, the program freezes.
(TeraTerm 5.3 on the remote site).
So my corrected code looks like this:
```
WRERR:	CALL	RSDMA		; RESET DMA TO NORMAL
   	MVI	A,CAN		; CANCEL
	CALL	SEND		; SENDER
 	CALL	SEND
 	CALL	SEND
```
Regards
Hubert

**Comment by wwarthen** [comment]
Thank you very much Hubert.  This explains some behavior I have seen in the
past.

I am not entirely surprised.  I have found several similar mistakes in
XModem 12.5 over the years.  The XModem source went through many changes by
many people over time.

I will make the fix in my source.

Thanks, Wayne

**Comment by wwarthen** [comment]
Corrected in commit 3470b9a.

Thank you @hubertushirsch.
